src/main/java/org/apache/sysml/parser/DMLTranslator.java

```diff
@@ -1272,6 +1272,10 @@ public void constructHops(StatementBlock sb)
 					if (!(target instanceof IndexedIdentifier)) {
 						//process right hand side and accumulation
 						Hop ae = processExpression(source, target, ids);
+						// if it is eval built-in func, add hop to output
+						if (ae instanceof NaryOp && ((NaryOp) ae).getOp().equals(OpOpN.EVAL)) {
+							output.add(ae);
+						}
```

Comment:
Why was this necessary and not sufficiently handled by the default case for setting the output properties?